### PR TITLE
[View Variables] Teleport Here

### PR DESCRIPTION
**Pull request by DeityLink**
Adds "Teleport Here" to the View Variables' drop down menu.
Teleports the mob/obj to where you are standing.
Similar to the Player Panel's "Get" button, however this one also works on objects.
Produces a "flash" like the one produced by admins spawning-in as Test Dummies.

here
chair_tp

### CODE REVIEW

**Comment by PJB3005** [comment]
:+1:

**Comment by DeityLink** [comment]
As was requested by @clusterfack added a Delete option.
If the atom is a mob with a player in it, confirmation will be asked.

File: code/datums/datumvars.dm [hunk 1]
```diff
@@ -704,6 +709,46 @@ client
 		src.holder.marked_datum = D
 		href_list["datumrefresh"] = href_list["mark_object"]
 
+	else if(href_list["teleport_here"])
+		if(!check_rights(0))	return
+
+		var/atom/A = locate(href_list["teleport_here"])
+		if(!isobj(A) && !ismob(A))
```
Comment (clusterfack) [inline code/datums/datumvars.dm]: You can have it typecast as atom/movable instead and check against that

File: code/datums/datumvars.dm [hunk 2]
```diff
@@ -704,6 +709,46 @@ client
...
+		if(!T)
+			usr << "You cannot teleport something into nullspace. Well I mean technically you can but that's not what that option is for."
+			return
+
+		if(isobj(A))
+			var/obj/O = A
+			O.loc = T
+		else if(ismob(A))
+			var/mob/M = A
+			M.loc = T
```
Comment (clusterfack) [inline code/datums/datumvars.dm]: Forcemove is the standard now

**Comment by clusterfack** [comment]
Thanks this'll be helpful

**Comment by IratePirate** [comment]
The more behind-the-scenes admin tools like this you improve and add, the more potential the serb has for great adminbus. Good on you.

**Comment by ComicIronic** [comment]
The flash should be optional, you should be able to delete anything (you already can with a verb) and you should cast as an atom/movable for teleport, rather than checking mob and obj.

**Comment by clusterfack** [comment]
You can del anything with the right click verb yes (the del menu itself is too vague to be useful), but inside objects or ones found through VV-ing around cannot be very easily.

**Comment by DeityLink** [comment]
@ComicIronic this option is mainly meant for adminbus when you want to move something to an area with players and with an effect to add some more credibility to an hypothetical "We teleported this item to your station -signed Nanotrasen" announcement.
If you don't want a flash you can just Varedit.

**Comment by ComicIronic** [comment]
I support being able to del from the VV menu, but there's no reason to restrict it.

File: code/datums/datumvars.dm [hunk 3]
```diff
@@ -267,6 +267,9 @@ client
 		if(ismob(D))
 			body += "<option value='?_src_=vars;mob_player_panel=\ref[D]'>Show player panel</option>"
 
+		if(istype(D,/atom))
```
Comment (ComicIronic) [inline code/datums/datumvars.dm]: /atom/movable

Comment (DeityLink) [inline code/datums/datumvars.dm]: k

**Comment by d3athrow** [comment]
I don't like the flash being not optional

**Comment by DeityLink** [comment]
You're the boss, here have some modularity:

* Added an option in the View Variables's drop down menu that lets admins teleport a mob/object to where they are standing.
* Admins can set the teleportation to be either flashy, with effects or even screen shaking, or stealthy with a fade-in. This preference is set in the Debug tab.
* Added another option that lets admins delete a mob/object from the View Variables menu.
* Added an admin spawnable "calm" Nar-Sie that by default doesn't move, doesn't consume atoms, and doesn't trigger Hell nor force a shuttle call. Mostly for debug and testing purposes.
* Plastic Flaps are now replaced by cult grilles by Nar-Sie (which are airtight when not broken).
* Fixed not being able to move through partially broken cult grilles

**Comment by d3athrow** [comment]
Frivilous addition that can be done easily with direct/global narrate

**Comment by DeityLink** [comment]
@d3athrow but now you can have the message appear at the exact time to teleport the atom in without having to juggle with clicks

**Comment by d3athrow** [comment]
BRBRBRBRBRBRBRBRBR

**Comment by PJB3005** [comment]
>Plastic Flaps are now replaced by cult grilles by Nar-Sie (which are airtight). 

ATOMICITY

**Comment by DeityLink** [comment]
but it's basically a bugfix! otherwise areas like cargo get depressurized if the shuttle isn't there.

**Comment by ComicIronic** [comment]
When we say "bugfixes don't have to be atomic", that means you can bundle lots of fixes together or you can fix while you change something.

It doesn't mean you can jam whatever bugfix into whatever PR!

**Comment by DeityLink** [comment]
You're right, still, we want PRs to be atomic so they can be more easily reviewed right, and this "bugfix" is only 3 lines long so, can you make an exception? :(

**Comment by d3athrow** [comment]
The atomic rule as far as bugfixes goes if you bundle a fix with a feature PR then that fix wont get merged until the feature part is ready as well

**Comment by DeityLink** [comment]
Well the feature part of this PR has been waiting for Final Decision for 3 days. 
I've added a Teleport To option as you requested on Steam.

**Pull request merged**